### PR TITLE
Implement data request tracking

### PR DESCRIPTION
**Pull request by arkid15r**
## Issue Number

#3361 

## Purpose/Implementation Notes

Add a command that gathers/shows stats based on a accession code list. 
The areas of interest include attempted/available experiments (total and broken down by source) as well as attempted/available samples (total and broken down by source).
The total number of jobs created includes all job types.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+import requests
+
+from data_refinery_common.logging import get_and_configure_logger
+from data_refinery_common.models import DownloaderJob, Experiment, ProcessorJob, SurveyJob
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--source-url",
```
Comment (davidsmejia) [inline foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py:14]:
```suggestion
            "--file",
```

File: foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            type=str,
+            help="The experiment accessions source file URL",
```
Comment (davidsmejia) [inline foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py:16]:
```suggestion
            help="A file listing accession codes. s3:// URLs only are accepted.",
```

File: foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        experiments_available = 0
+        experiments_available_by_source = Counter()
+        jobs_created = 0
+        jobs_created_by_source = Counter()
+        samples_attempted = 0
+        samples_attempted_by_source = Counter()
+        samples_available = 0
+        samples_available_by_source = Counter()
+
+        for accession in self.get_accessions(options["source_url"]):
```
Comment (davidsmejia) [inline foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py:32]:
```suggestion
        for accession in self.get_accessions(options["file"]):
```

File: foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py (hunk 1 above)
Comment (davidsmejia) [reply foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py:14]: This is just to make it consistent with other management commands.

File: foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            output.append(f"Total jobs: {total_jobs_created}")
+        else:
+            output.append("No experiments found")
+
+        print("\n".join(output))
+
+    @staticmethod
+    def get_accessions(url):
+        """Generates source experiment accessions."""
+        return (line.strip() for line in requests.get(url).text.split("\n") if line.strip())
```
Comment (davidsmejia) [inline foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py:106]: Could we test for protocol and then use `boto3` as done here: https://github.com/AlexsLemonade/refinebio/blob/dbc64643b43b3b8e86a63e91d10a5ddfb043ca2e/foreman/data_refinery_foreman/surveyor/management/commands/unsurvey.py#L242-L254

**Review by davidsmejia — CHANGES_REQUESTED** [review]
Looking good, could we just add support for s3:// urls

File: foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py (hunk 4 above)
Comment (arkid15r) [reply foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py]: Done.

File: foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py (hunk 3 above)
Comment (arkid15r) [reply foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py]: Renamed.

File: foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py (hunk 2 above)
Comment (arkid15r) [reply foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py]: Reworked to accept s3:// files only

File: foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py (hunk 1 above)
Comment (arkid15r) [reply foreman/data_refinery_foreman/foreman/management/commands/track_data_request.py]: Done.

**Review by arkid15r — COMMENTED** [review]
I've updated PR based on your comments, PTAL

**Review by davidsmejia — APPROVED** [review]
LGTM

**Pull request merged**